lte/gateway/deploy/agw_install_ubuntu.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+# Copyright 2021 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Setting up env variable, user and project path
+MAGMA_USER="magma"
+AGW_INSTALL_CONFIG="/etc/systemd/system/multi-user.target.wants/agw_installation.service"
+AGW_SCRIPT_PATH="/root/agw_install_ubuntu.sh"
+DEPLOY_PATH="/home/$MAGMA_USER/magma/lte/gateway/deploy"
+SUCCESS_MESSAGE="ok"
+NEED_REBOOT=0
+WHOAMI=$(whoami)
+MAGMA_VERSION="${MAGMA_VERSION:-v1.5}"
+CLOUD_INSTALL="cloud"
+GIT_URL="${GIT_URL:-https://github.com/magma/magma.git}"
+
+
+echo "Checking if the script has been executed by root user"
+if [ "$WHOAMI" != "root" ]; then
+  echo "You're executing the script as $WHOAMI instead of root.. exiting"
+  exit 1
+fi
+
+wget https://raw.githubusercontent.com/magma/magma/"$MAGMA_VERSION"/lte/gateway/deploy/agw_pre_check_ubuntu.sh
+if [[ -f ./agw_pre_check_ubuntu.sh ]]; then
+  chmod 644 agw_pre_check_ubuntu.sh && bash agw_pre_check_ubuntu.sh
+  while true; do
+      read -r "Do you accept those modifications and want to proceed with magma installation?(y/n)" yn
+      case $yn in
+          [Yy]* ) break;;
+          [Nn]* ) exit;;
+          * ) echo "Please answer yes or no.";;
+      esac
+  done
+else
+  echo "agw_pre_check_ubuntu.sh is not available in your version"
+fi
+
+echo "Checking if Ubuntu is installed"
+if ! grep -q 'Ubuntu' /etc/issue; then
+  echo "Ubuntu is not installed"
+  exit 1
+fi
+
+echo "Making sure $MAGMA_USER user is sudoers"
+if ! grep -q "$MAGMA_USER ALL=(ALL) NOPASSWD:ALL" /etc/sudoers; then
+  apt install -y sudo
+  adduser --disabled-password --gecos "" $MAGMA_USER
+  adduser $MAGMA_USER sudo
+  echo "$MAGMA_USER ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+fi
+
+echo "Need to check if both interfaces are named eth0 and eth1"
+INTERFACES=$(ip -br a)
+if [[ $1 != "$CLOUD_INSTALL" ]] && ( [[ ! $INTERFACES == *'eth0'*  ]] || [[ ! $INTERFACES == *'eth1'* ]] || ! grep -q 'GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"' /etc/default/grub); then
+  # changing intefaces name
+  sed -i 's/GRUB_CMDLINE_LINUX=""/GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"/g' /etc/default/grub
```

Comment:
Do we still need that with netplan?

Reply:
we are not using netplan, but even with netplan we need to rename interfaces.